src/main/java/de/tum/in/www1/artemis/service/QuizExerciseService.java

```diff
@@ -97,16 +116,19 @@ private void updateResultsOnQuizChanges(QuizExercise quizExercise) {
     /**
      * @param quizExercise         the changed quiz exercise from the client
      * @param originalQuizExercise the original quiz exercise (with statistics)
+     * @param files                the files that were uploaded
      * @return the updated quiz exercise with the changed statistics
      */
-    public QuizExercise reEvaluate(QuizExercise quizExercise, QuizExercise originalQuizExercise) {
-
+    public QuizExercise reEvaluate(QuizExercise quizExercise, QuizExercise originalQuizExercise, @Nonnull List<MultipartFile> files) throws IOException {
```

Comment:
Why does a reevaluation need the files? Shouldn't this only change the scores when another answer option is marked as valid or a question is deactivated?

Reply:
Nope, you can change segments in the client. Feel free to try it out.

I was surprised myself, but it's intended functionality.